src/lib.rs

```diff
@@ -284,9 +288,34 @@ cfg_if! {
 /// In general, `getrandom` will be fast enough for interactive usage, though
 /// significantly slower than a user-space CSPRNG; for the latter consider
 /// [`rand::thread_rng`](https://docs.rs/rand/*/rand/fn.thread_rng.html).
+#[inline]
 pub fn getrandom(dest: &mut [u8]) -> Result<(), Error> {
-    if dest.is_empty() {
-        return Ok(());
-    }
-    imp::getrandom_inner(dest)
+    // SAFETY: The `&mut MaybeUninit<_>` reference doesn't escape.
+    getrandom_uninit_slice(unsafe { slice_as_uninit_mut(dest) }).map(|_| ())
```

Comment:
I'll remove the mapping, but I am going to keep the `unsafe { }` inline, because the safety assertion is clearer when the `&mut MaybeUninit<_>` is temporary.

Reply:
Looks reasonable to me.